test/tools/runner/filters/auth_filter.js

```
@@ -0,0 +1,36 @@
+'use strict';
+
+/**
+ * Filter for authorization enabled or disabled on the server
+ *
+ * example:
+ * metadata: {
+ *    requires: {
+ *      auth: 'enabled' | 'disabled'
+ *    }
+ * }
+ */
+class AuthFilter {
+  constructor() {
+    this.isAuthEnabled = process.env.AUTH === 'auth';
+  }
+
+  filter(test) {
+    if (!test.metadata) return true;
+    if (!test.metadata.requires) return true;
+    if (!test.metadata.requires.auth) return true;
+
+    const auth = test.metadata.requires.auth;
+
+    if (auth === 'enabled') {
+      return this.isAuthEnabled;
+    } else if (auth === 'disabled') {
+      return !this.isAuthEnabled;
+    }
+
+    // defaults to 'disabled' because it's gotta default to something.
+    return true;
```

could we throw instead so that we catch typos/unsupported values?